### PR TITLE
Make it so the RazorDirectiveCompletionProvider doesn't load extra as…

### PR DESCRIPTION
**Pull request by NTaylorMullen**
…semblies in C#.

- Added a layer of indirection to allow for proper Lazy loading of Razor and future async completion specific assemblies.

### CODE REVIEW

File: src/Microsoft.VisualStudio.Editor.Razor/RazorDirectiveCompletionProvider.cs [hunk 1]
```diff
@@ -207,4 +199,40 @@ protected virtual bool TryGetRazorSnapshotPoint(CompletionContext context, out S
             return false;
         }
     }
+
+    // These types are only for this class to provide indirection for assembly loads.
+    internal abstract class CompletionProviderDependencies
```
Comment (NTaylorMullen) [inline src/Microsoft.VisualStudio.Editor.Razor/RazorDirectiveCompletionProvider.cs]: @rynowak if you know of a better way to do this I'm all ears 😉

File: src/Microsoft.VisualStudio.Editor.Razor/RazorDirectiveCompletionProvider.cs [hunk 2]
```diff
@@ -111,13 +103,6 @@ public override Task ProvideCompletionsAsync(CompletionContext context)
                 return Task.CompletedTask;
             }
 
-            if (!_textBufferProvider.TryGetFromDocument(context.Document, out var textBuffer) ||
-                !_asyncCompletionBroker.IsCompletionSupported(textBuffer.ContentType))
```
Comment (NTaylorMullen) [inline src/Microsoft.VisualStudio.Editor.Razor/RazorDirectiveCompletionProvider.cs]: We shouldn't be the one loading extra IAsyncCompletion bits if we're not operating on a Razor file.

File: src/Microsoft.VisualStudio.Editor.Razor/RazorDirectiveCompletionProvider.cs (hunk 1 above)
Comment (rynowak) [reply src/Microsoft.VisualStudio.Editor.Razor/RazorDirectiveCompletionProvider.cs]: No, this seems about right.

**Comment by NTaylorMullen** [comment]
> Should we have a work item to undo this once modern completion lands?

Have one already in VSTS that everyone always gets bugged about 😄.

**Pull request merged**